Topics/Tech_Stacks/CSS.md

```diff
@@ -26,6 +27,62 @@ Because of the existence of ids and classes in selectors, often multiple CSS dec
 
 [This website](https://wattenberger.com/blog/css-cascade) gives a detailed breakdown of CSS Cascade and its tiers. 
 
+## CSS Selector Specificity:
+
+CSS specificity is the set of rules and calculations that determine which CSS selectors are executed in the CSS Cascade. Understanding this crucial in improving the fluency of a programmer's ability to write functional CSS, since this concept can save many hours of troubleshooting.
+
+#### Specificity Hierarchy
+The selector with the highest specificity is chosen to execute on the CSS Cascade. There are 3 selector categories, from most specific to least specific, shown below:
+
+1. **ID**
+2. **Classes and Pseudo-classes**
+3. **Elements and Pseudo-elements**
+
+#### How is specificity calculated?
+All selectors have 3 values, where each value represents the number of instances it contains of the 3 selector categories. For every pair of selectors, the more specific selector is determined by comparing which value is greater from their respective selectors, from left to right.
+
+For example:
+
+
+`#im-an-id { ... }  /* IDs=1, Classes=0, Elements=0 */` -> Specificity = 100
+
+`.im-a-class { ... }  /* IDs=0, Classes=1, Elements=0 */` -> Specificity = 010
+
+`p { ... }  /* IDs=0, Classes=0, Elements=1 */` -> Specificity = 001
+
+These selectors are ordered from most specific to least specific.
+
+Suppose we introduce a fourth selector:
+
+`#im-the-best-id.im-another-class.im-yet-another-class { ... }  /* IDs=1, Classes=2, Elements=0 */` -> Specificity = 120
+
+This selector would have the highest specificity of all the previous examples. It has the same number of IDs as the first selector, but has 2 more classes, making it more specific.
+
+N.B. that selector combinators have no impact on the specificity calculation, and inline styles of an element have the highest specificity regardless of the categories above.
+
+#### Troubleshooting
```

Comment:
This is a really great section, I was wondering if there were any online resources to add to this such as an online calculator that would do the specificity calculations for you if you were stuck debugging? I think it would be a great resource to have for a developer!

Reply:
I have limited experience using a specificity calculator, but I can understand why it can be tremendously useful in troubleshooting css. I'll add it, thanks